### PR TITLE
feat: add test 6.3.11 for CSAF 2.1, regex refactor, supplemental test cases

### LINKED ISSUE #279: Implement 6.3.11 Usage of V as Version Indicator for CSAF 2.1
Already works for 2.0.
ToDo:
- [ ] check if current implementation works for 2.1
- [ ] check and extend supplementary test cases for 2.1
- [ ] update README
- [ ] update validation.rs

### PR DESCRIPTION
**Pull request by peinjoh**
This PR:
* adds 6.3.11 for CSAF 2.1
* moves from a trivial regex to a equal manual string check
* adds supplemental test cases
  * S01: V4.2 (uppcase V)
  * S02: multiple branches, alternating findings
  * S03: stacked categories, alternating findings (violates 6.1.57)
  * S11: vAlpha
  * S12: v4.2, but in branch category architecture
  
Resolves #279

### CODE REVIEW

File: csaf-rs/src/validations/test_6_3_11.rs [hunk 1]
```diff
@@ -1,9 +1,12 @@
...
 use crate::validation::{TestFinding, TestFindingData};
-use regex::Regex;
-use std::sync::LazyLock;
 
-static V_AS_VERSION_INDICATOR_REGEX: LazyLock<Regex> = LazyLock::new(|| Regex::new(r"^[vV][0-9].*$").unwrap());
+/// To implement this test it is deemed sufficient that the value of name does not match the following regex:
+/// `^[vV][0-9].*$`. This function implements this trivial regex using regular string operations, which is significantly faster.
+fn is_version_with_v_indicator(version: &str) -> bool {
+    let mut chars = version.chars();
+    matches!(chars.next(), Some('v') | Some('V')) && matches!(chars.next(), Some(c) if c.is_ascii_digit())
```
[the remark this answers is not in the document]
Comment (peinjoh) [reply csaf-rs/src/validations/test_6_3_11.rs:9]: That is true, but I would argue that the currently implementation is still more "in spirit" of the test. The previous regex is from an informative comment.

> For each element of type $['$defs'].branches_t with category of product_version it SHALL be tested that the value of name does not start with v or V before the version.

> To implement this test it is deemed sufficient that the value of name does not match the following regex:
^[vV][0-9].*$

I would argue that the test should flag any branch name that starts with case-insensitive v + a numeric char, even if what comes after is a newline.

Very much an edge case, but should be brought up to TC. The property does not forbid newlines.

[the remark this answers is not in the document]
Comment (peinjoh) [reply csaf-rs/src/validations/test_6_3_11.rs:9]: Will be brought up here: #928

File: type-generator/assets/tests/csaf_2.0/testcases.json [hunk 2]
```diff
@@ -687,6 +687,34 @@
...
+          "name": "informative/csaf-rs_csaf-csaf_2_0-6-3-11-s01.json",
+          "valid": true
+        },
+        {
+          "name": "informative/csaf-rs_csaf-csaf_2_0-6-3-11-s02.json",
+          "valid": true
+        },
+        {
+          "name": "informative/csaf-rs_csaf-csaf_2_0-6-3-11-s03.json",
+          "valid": false
```
[the remark this answers is not in the document]
Comment (peinjoh) [reply type-generator/assets/tests/csaf_2.0/testcases.json]: Valid.

**Review by Gronner — APPROVED** [review]
LGTM
